source/common/upstream/upstream_impl.cc

```diff
@@ -506,6 +508,8 @@ void ClusterImplBase::onPreInitComplete() {
       pending_initialize_health_checks_ += host_set->hosts().size();
     }
 
+    sds_init_manager_.initialize([]() -> void {});
```

Comment:
This is wrong. 

How about split the ClusterImplBase::onPreInitComplete into following two functions:

ClusterImplBase::onPreInitComplete() {
   sds_init_manager_.initialize( [this]() { onSdsInitDone(); });
}

ClusterImplBase::onSdsInitDone() {
// exact same code as onPreInitComplete
}